nilearn/plotting/tests/test_matrix_plotting.py

```diff
@@ -274,6 +274,18 @@ def test_show_contrast_matrix(tmp_path):
     assert (tmp_path / "contrast.pdf").exists()
 
 
+def test_show_contrast_matrix_axes():
+    pytest.importorskip("matplotlib", minversion="3.5.0")
```

Comment:
is this because constrained was introduced in matplotlib 3.5 ?

Reply:
Constrained layout was available before then but with a different kwarg like `use_constrained_layout=True`. 3.5 is when they settled on the cleaner `layout="constrained"` API. Didn't seem worth triaging just to get it to run on your oldest supported version (3.3).

FWIW, matplotlib 3.5 is [out of the SPEC0 window](https://scientific-python.org/specs/spec-0000/) (and even 3.6 is almost there!) so you could consider adopting that spec and/or bumping your minimum matplotlib requirement to at least 3.5. I wouldn't make the switch to constrained layout until you do, otherwise you'd have to triage kwargs based on mpl version which is a pain (and ugly in examples if you want those to be backward compatible).

Reply:
> FWIW, matplotlib 3.5 is [out of the SPEC0 window](https://scientific-python.org/specs/spec-0000/) (and even 3.6 is almost there!) so you could consider adopting that spec and/or bumping your minimum matplotlib requirement to at least 3.5. I wouldn't make the switch to constrained layout until you do, otherwise you'd have to triage kwargs based on mpl version which is a pain (and ugly in examples if you want those to be backward compatible).

We talked about this internally and deliberately decided to lag behind SPEC0 for now:

https://github.com/nilearn/nilearn/issues/4425#issuecomment-2182070479